### PR TITLE
Assumed token duration check

### PR DESCRIPTION
**Pull request by salunn**

### CODE REVIEW

File: lib/yle/aws/role.rb [hunk 1]
```diff
@@ -1,16 +1,15 @@
-require 'aws-sdk-core'
+require 'aws-sdk-iam'
```
Comment (tmatilai) [inline lib/yle/aws/role.rb]: There is nothing requiring the IAM library on this file, so the core should be enough.

File: lib/yle/aws/role.rb [hunk 2]
```diff
@@ -36,15 +35,17 @@ def self.default_role_name
       end
 
       def self.default_duration
-        config['defaults']['duration'] || DEFAULT_DURATION
+        default_token_duration
```
Comment (tmatilai) [inline lib/yle/aws/role.rb]: With this change the whole method is IMHO not needed any more

File: lib/yle/aws/role/token_duration.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+require 'aws-sdk-iam'
+require 'yle/aws/role/config'
+
+module Yle
+  module AWS
+    class Role
```
Comment (tmatilai) [inline lib/yle/aws/role/token_duration.rb:6]: This should be in its own namespace, i.e. a `TokenDuration` class or module, which is then used from the `Role` class.

File: lib/yle/aws/role/token_duration.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      # Default duration in seconds when assuming a role
+      DEFAULT_DURATION = 900
+      #AWS gives lower bound for max age as 1 hour
```
Comment (tmatilai) [inline lib/yle/aws/role/token_duration.rb]: Missing space after `#`. And maybe an empty line before.

File: lib/yle/aws/role/token_duration.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      SHORTEST_MAXIMUM_ALLOWED = 3600
+
+      #private
+
+      def self.default_token_duration
+        config['defaults']['duration'] || DEFAULT_DURATION
+      end
+
+      def get_validated_duration(account, role_name, duration)
```
Comment (tmatilai) [inline lib/yle/aws/role/token_duration.rb]: In ruby the `get_` prefixes are not recommended

File: lib/yle/aws/role/token_duration.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      def get_max_duration_for_role(assumed_role, role_name)
+        Aws::IAM::Client.new({credentials: assumed_role})
+          .get_role({role_name: role_name})
+          .role.max_session_duration
+      end
+
+      def cap_duration_at_max_allowed(max_duration, duration)
+        if duration > max_duration
+          puts "Longest allowed duration is #{max_duration} seconds, using that as duration."
+          puts "If you require longer tokens for the role, contact your AWS administrators."
```
Comment (tmatilai) [inline lib/yle/aws/role/token_duration.rb]: These messages should go to `STDERR`

**Review by tmatilai — COMMENTED** [review]
Great feature, but some style issues below. You should also run `rubocop -SD` and see what it complains.

File: lib/yle/aws/role/token_duration.rb (hunk 3 above)
Comment (salunn) [reply lib/yle/aws/role/token_duration.rb:6]: I put it in as a nested `TokenDuration` class, since it feels like the two are linked closely enough.

**Pull request closed without merging**